addons/mail/static/src/js/form_renderer.js

```diff
@@ -1,77 +1,151 @@
  */
 FormRenderer.include({
+    async on_attach_callback() {
+        this._super(...arguments);
+        // When the view is re-attached (_isInDom is true) and the chatter component already exists
+        // the chatter component needs to be retargeted in the DOM by force mounting.
+        if (this._chatterComponent && this._isInDom) {
+            await this._forceMountChatterComponent();
+        }
+    },
+    on_detach_callback() {
+        this._super(...arguments);
+        // When the view is detached, we totally delete chatter state from store and chatter
+        // component to avoid any problem when view will be reattached
+        if (this._chatterComponent) {
+            this._deleteChatter();
+        }
+    },
     /**
      * @override
      */
-    init: function (parent, state, params) {
-        this._super.apply(this, arguments);
+    init(parent, state, params) {
+        this._super(...arguments);
+        this.env = this.call('messaging', 'getMessagingEnv');
         this.mailFields = params.mailFields;
-        this.chatter = undefined;
+        // Do not load chatter in form view dialogs
+        this._preventChatterLoad = params.fromFormViewDialog;
+        this._chatterComponent = undefined;
+        this._chatterLocalId = undefined;
+        /**
+         * Determine if the form view has a chatter
+         * (i.e. if a div .oe-chatter is detected in the arch)
+         * @type {boolean}
+         * @private
+         */
+        this._hasChatter = false;
     },
-
     //--------------------------------------------------------------------------
     // Public
     //--------------------------------------------------------------------------
-
-    /**
-     * Updates the chatter area with the new state if its fields has changed
-     *
-     * @override
-     */
-    confirmChange: function (state, id, fields) {
-        if (this.chatter) {
-            var chatterFields = ['message_attachment_count'].concat(_.values(this.mailFields));
-            var updatedMailFields = _.intersection(fields, chatterFields);
-            if (updatedMailFields.length) {
-                this.chatter.update(state, updatedMailFields);
-            }
+    destroy() {
+        this._super(...arguments);
+        if (this._hasChatter) {
+            this._deleteChatter();
         }
-        return this._super.apply(this, arguments);
     },
-
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
-
     /**
-     * Overrides the function that renders the nodes to return the chatter's $el
-     * for the 'oe_chatter' div node.
+     * @private
+     */
+    async _createChatter() {
+        const chatterLocalId = await this.env.store.dispatch('createChatter', {
+            initialThreadId: this.state.res_id,
+            initialThreadModel: this.state.model,
+        });
```

Comment:
`awaiting` store dispatch in a component is risky, so I would remove that.
Could let `createChatter` return `chatterLocalId` synchronously, even though it asynchronously fetches thread messages and attachments in background.